mcc/health/cmd/create.go

```diff
@@ -29,87 +31,51 @@ var createCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
 		message := make(map[string]interface{})
 		out := []string{}
+		// Select backend to prepare request
 		switch {
 		case strings.Contains(endpoint, "healthchecks.io"):
 			check = health.NewCheck()
-			check.SetAPIKey(apikey)
-			if schedule != "" {
-				message["schedule"] = schedule
-				out = append(out, schedule)
-			}
-			if name != "" {
-				message["name"] = name
-				out = append(out, name)
-			}
-			if tags != "" {
-				message["tags"] = tags
-				out = append(out, tags)
-			}
-			req, err := check.Create(endpoint, message)
-			if err != nil {
-				log.Fatal(err)
-			}
-			res, err := http.DefaultClient.Do(req)
-			if err != nil {
-				log.Fatal(err)
-			}
-			m, err := health.ParseResponse(res.Body)
-			if err != nil {
-				log.Fatal(err)
-			}
+			message = health.SetMessage(schedule, name, tags)
+		case strings.Contains(endpoint, "cronitor.io"):
+			check = cronitor.NewCheck()
+			message = cronitor.SetMessage(schedule, name, tags, email)
+		default:
+			log.Fatal("Unrecognized provider")
+		}
+		// backend independent processing
+		check.SetAPIKey(apikey)
+		req, err := check.Create(endpoint, message)
+		if err != nil {
+			log.Fatal(err)
+		}
+		res, err := http.DefaultClient.Do(req)
+		if err != nil {
+			log.Fatal(err)
+		}
+		m := make(map[string]interface{})
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = json.Unmarshal(body, &m)
```

Comment:
Isn't there in go a more generic way of capturing errors/exceptions, so you can catch several exceptions in just one block and print the result instead of having to deal error by error?

Reply:
Not in the language. There are some proposed patterns to reduce the verbosity of error handling and those mainly rely on functions that wrap the calls and handle the error internally. In this case it could work (as I handle it the same every time for simplicity), but I think it makes the code less readable.

Also, better handling of errors would be necessary to consider the tool for a broader use than our current minimal itch, and that would break the unified approach mentioned in the previous paragraph.

I'm open to suggestions if you encounter a better method for handling it in this case 😄